rayon-core/src/lib.rs

```diff
@@ -359,3 +360,41 @@ impl fmt::Debug for Configuration {
          .finish()
     }
 }
+
+/// Provides the calling context to a closure called by `join_context`.
+#[derive(Debug)]
+pub struct FnContext {
+    migrated: bool,
+
+    /// disable `Send` and `Sync`, just for a little future-proofing.
+    _marker: PhantomData<*mut ()>,
+}
+
+impl Default for FnContext {
```

Comment:
Do we really want this to support `default`? That means anybody can make one...?

Reply:
I thought I needed a way for the iterator bridge functions to create one -- which is as good as "anybody" as far as crate privacy goes -- but I just modified those to extract the `migrated` bool to pass around instead.